src/rdinterval.h

```diff
@@ -95,6 +96,19 @@ static RD_INLINE RD_UNUSED void rd_interval_reset (rd_interval_t *ri) {
         ri->ri_backoff = 0;
 }
 
+/**
+ * Reset the interval to 'now'. If now is 0, the time will be gathered
+ * automatically.
+ */
+static RD_INLINE RD_UNUSED void rd_interval_reset_to_now (rd_interval_t *ri,
```

Comment:
as you suggest was my first thought too. I specifically did it the way I have because i think it makes the api clearer. if done as you suggest, the now parameter would need to be tri-state (one value to say reset is immediate, another to say use current time, another to specify a time). also, it makes the comments more long winded and difficult to interpret in a number of places (the reset method, and the rd_interval0 method which explains when the interval will trigger immediately). as is, they are straightforward.

Reply:
Sounds good :+1: